packages/zendesk-adapter/src/change_validators/guide_disabled.ts

```diff
@@ -14,72 +14,52 @@
+   const relevantInstances = changes
+     .filter(isInstanceChange)
+     .filter(isAdditionChange)
+     .map(getChangeData)
+     .filter(instance => GUIDE_TYPES_TO_HANDLE_BY_BRAND.includes(instance.elemID.typeName))
+   if (_.isEmpty(relevantInstances)) {
+     return []
+   }
+   if (elementSource === undefined) {
+     log.error('Failed to run guideDisabledValidator because no element source was provided')
+     return []
+   }
 
-export const guideDisabledValidator: ChangeValidator = async (changes, elementSource) => {
-  const relevantInstances = changes
-    .filter(isInstanceChange)
-    .filter(isAdditionChange)
-    .map(getChangeData)
-    .filter(instance => GUIDE_TYPES_TO_HANDLE_BY_BRAND.includes(instance.elemID.typeName))
-  if (_.isEmpty(relevantInstances)) {
-    return []
-  }
-  if (elementSource === undefined) {
-    log.error('Failed to run guideDisabledValidator because no element source was provided')
-    return []
-  }
+   const brandsInstances = await awu(await elementSource.list())
+     .filter(id => id.typeName === BRAND_TYPE_NAME)
+     .map(id => elementSource.get(id))
+     .filter(isInstanceElement)
+     .toArray()
 
-  const BrandsByBrandId = Object.fromEntries((await awu(await elementSource.list())
-    .filter(id => id.typeName === BRAND_TYPE_NAME)
-    .map(id => elementSource.get(id))
-    .filter(isInstanceElement)
-    .toArray())
-    .map(instance => [instance.elemID.getFullName(), instance]))
-  if (_.isEmpty(BrandsByBrandId)) {
-    return []
-  }
+   const brandsWithGuide = new Set(getBrandsForGuide(brandsInstances, fetchConfig)
+     .map(brand => brand.elemID.getFullName()))
 
-  const brandsWithoutGuideByInstanceId = getBrandsWithoutGuideByInstanceId(relevantInstances, BrandsByBrandId)
+   if (brandsWithGuide === undefined) {
+     return []
+   }
 
-  return relevantInstances
-    .filter(instance => instance.elemID.getFullName() in brandsWithoutGuideByInstanceId)
-    .map(instance => ({
-      elemID: instance.elemID,
-      severity: 'Error',
-      message: 'Cannot add this element because help center is not enabled for its associated brand.',
-      detailedMessage: `please enable help center for brand "${brandsWithoutGuideByInstanceId[instance.elemID.getFullName()].elemID.name}" in order to add this element.`,
-    }))
-}
+   return relevantInstances
+     .filter(instance => isReferenceExpression(instance.value.brand))
+     .filter(instance => !brandsWithGuide.has(instance.value.brand.elemID.getFullName()))
+     .map(instance => ({
+       elemID: instance.elemID,
+       severity: 'Error',
+       message: 'Cannot add this element because help center is not enabled for its associated brand, or the brand itself may not be enabled in the configuration.',
+       detailedMessage: `please enable help center for brand "${instance.value.brand.elemID.name}" or enable the brand in the configuration in order to add this element.`,
```

Comment:
nit:
```suggestion
       detailedMessage: `Please enable help center for brand "${instance.value.brand.elemID.name}" or enable the brand in the configuration in order to add this element.`,
```

also, I think we should be more clear about the reason the element is not deployable.. we can have a different message for each scenario because it requires 2 different actions from the user (enable brand in guide or change brand config in salto)